cpp/Djikstra.cpp

```diff
@@ -0,0 +1,123 @@
+#include<iostream>
+#include<set>
+#include<list>
+#include<algorithm>
+using namespace std;
+
+typedef struct nodes {
+   int dest;
+   int cost;
+}node;
+
+class Graph {
+   int n;
+   list<node> *adjList;
+   private:
+      void showList(int src, list<node> lt) {
+         list<node> :: iterator i;
+         node tempNode;
+
+         for(i = lt.begin(); i != lt.end(); i++) {
+            tempNode = *i;
+            cout << "(" << src << ")---("<<tempNode.dest << "|"<<tempNode.cost<<") ";
+         }
+         cout << endl;
+      }
+   public:
+      Graph() {
+         n = 0;
+      }
+
+      Graph(int nodeCount) {
+         n = nodeCount;
+         adjList = new list<node>[n];
```

Comment:
Memory leak: replace with vector<list<node>> or better yet (for improved perf), vector<deque<node>>